src/filter/blur.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+import Image from '../image';
+
+// first release of mean filter
+export default function meanFilter(k){
+
+    if(this.components > 1){
+        throw new Error('Only support for image in gray scale');
+    }
+
+    if(k < 1){
+        throw new Error('k must be grater than 0');
+    }
+
+    //mean filter do not is in place
+    var newImage = Image.createFrom(this, {
+        kind: {
+            components: 1,
+            alpha: this.alpha,
+            bitDepth: this.bitDepth,
+            colorModel: null
+        }
+    });
+
+    /*
+    Example of 3x3 kernel:
+    1   1   1
+    1   1   1
+    1   1   1
+    */
+    var n = 2*k + 1;
+    var kernel = new Int8Array(n*n);
```

Comment:
We could define once for ever  `let size = n*n;`

Please note that there is now a test rule that forbids to use `var` and you should replace all `var`by `let` or `const`. `const`are non modifiable variables. The difference  between `let`and `var`is that `let`is block-scoped. If you define a `var`in a loop for example it is available outside the loop in javascript in contrast with other languages and can lead to unexpected behaviour.